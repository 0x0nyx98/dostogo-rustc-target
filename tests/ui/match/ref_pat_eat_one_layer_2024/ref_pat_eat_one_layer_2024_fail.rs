//@ edition: 2024
//@ compile-flags: -Zunstable-options
#![allow(incomplete_features)]
#![feature(ref_pat_eat_one_layer_2024)]

pub fn main() {
    if let Some(&mut Some(&_)) = &Some(&Some(0)) {
        //~^ ERROR: mismatched types
    }
    if let Some(&Some(&_)) = &Some(&mut Some(0)) {
        //~^ ERROR: mismatched types
    }
}
