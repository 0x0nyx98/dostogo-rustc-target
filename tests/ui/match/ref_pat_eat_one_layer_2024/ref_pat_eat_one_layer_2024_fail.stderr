error[E0308]: mismatched types
  --> $DIR/ref_pat_eat_one_layer_2024_fail.rs:7:17
   |
LL |     if let Some(&mut Some(&_)) = &Some(&Some(0)) {
   |                 ^^^^^^^^^^^^^    --------------- this expression has type `&Option<&Option<{integer}>>`
   |                 |
   |                 types differ in mutability
   |
   = note:      expected reference `&Option<{integer}>`
           found mutable reference `&mut _`

error[E0308]: mismatched types
  --> $DIR/ref_pat_eat_one_layer_2024_fail.rs:10:23
   |
LL |     if let Some(&Some(&_)) = &Some(&mut Some(0)) {
   |                       ^^     ------------------- this expression has type `&Option<&mut Option<{integer}>>`
   |                       |
   |                       expected integer, found `&_`
   |
   = note:   expected type `{integer}`
           found reference `&_`

error[E0308]: mismatched types
  --> $DIR/ref_pat_eat_one_layer_2024_fail.rs:14:27
   |
LL |         let _: &mut u32 = x;
   |                --------   ^ types differ in mutability
   |                |
   |                expected due to this
   |
   = note: expected mutable reference `&mut u32`
                      found reference `&{integer}`

error[E0308]: mismatched types
  --> $DIR/ref_pat_eat_one_layer_2024_fail.rs:17:23
   |
LL |     if let Some(&Some(&x)) = Some(&Some(&mut 0)) {
   |                       ^^     ------------------- this expression has type `Option<&Option<&mut {integer}>>`
   |                       |
   |                       types differ in mutability
   |
   = note: expected mutable reference `&mut {integer}`
                      found reference `&_`
help: consider removing `&` from the pattern
   |
LL |     if let Some(&Some(x)) = Some(&Some(&mut 0)) {
   |                       ~

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0308`.
